internal/ethapi/api.go

```diff
@@ -752,10 +752,11 @@ func addEthCompatibilityFields(ctx context.Context, response map[string]interfac
 		var gasLimit uint64
 		var err error
 
-		// For mainnet, alfajores and baklava we have a set of hardcoded values derived from historical state that we can use.
-		v := params.PreGingerbreadNetworkGasLimits[b.ChainConfig().ChainID.Uint64()]
-		if v != nil {
-			gasLimit = v.Limit(header.Number)
+		// For mainnet, alfajores and baklava we have a set of hardcoded values derived from historical state that we can
+		// use, note ChainID might be unset, so we need to account for that.
```

Comment:
In which case is chainId unset?

Reply:
`cmd/geth.TestCustomGenesis` explicitly checks for unset chainID's, see [here](https://github.com/celo-org/celo-blockchain/blob/3a9fda2314604e7533cfc1086c3b1512c78071ae/cmd/geth/genesis_test.go#L40)